libs/networking/http_helpers/http_helpers.ml

```diff
@@ -25,12 +25,20 @@ open Cohttp
 (*****************************************************************************)
 let client_ref : (module Cohttp_lwt.S.Client) option ref = ref None
 
-module Make (Lwt_wrapper : sig
+(* We use a functor here so that we can pass in what we use for the Lwt runtime. *)
+(* This is platform dependent (JS vs Unix), so we can't just choose one *)
+(* We could use a ref like above, but this doesn't need to be decided at runtime, only *)
+(* at build, so I don't want to open the door to being able to change it at runtime *)
```

Comment:
But why not removing the functor and simply calls directly Lwt_platform.run?
The whole point of those virtual modules is that you don't have to functorize the code! By magic on your platform Lwt_platform will be been linked to the right one.

Reply:
In packages, you must choose an implementation of a virtual module, you can't let the thing using the package decide, I'll add a comment explaining this. Dune gets angry